src/allencell_ml_segmenter/thresholding/thresholding_service.py

```diff
@@ -64,32 +65,12 @@ def __init__(
             self._save_thresholded_images,
         )
 
-        self._viewer.subscribe_layers_change_event(
-            function=self._update_original_layers
-        )
-
     def _handle_thresholding_error(self, error: Exception) -> None:
         show_info("Thresholding failed: " + str(error))
 
     def _on_threshold_changed(self, _: Event) -> None:
-        # if we havent thresholded yet, keep track of original layers.
-        # need to check this on first threshold change, since user can add images
-        # between finishing prediction and starting thresholding
-        # if they are using images from a directory.
-        original_layers: Optional[OrderedDict[str, np.ndarray]] = (
-            self._thresholding_model.get_original_layers()
-        )
-        if original_layers is None:
-            self._thresholding_model.set_original_layers(
-                self._viewer.get_layers()
-            )
-
-        # Get layers to threshold.
-        # if there are segmentations displayed in the viewer, only threshold those images.
-        layers_to_threshold: OrderedDict[str, np.ndarray] = (
-            self._thresholding_model.get_layers_to_threshold(
-                self._main_model.are_predictions_in_viewer()
-            )
+        segmentation_labels: list[LabelsLayer] = (
```

Comment:
Do we need to copy this to local state on threshold changes?  Couldn't users of `segmentation_labels` just call `self._viewer.get_all_segmentation_labels()`?